dynamic_instrumentation/src/instrument_memory.rs

```diff
@@ -332,14 +380,21 @@ impl<'a, 'tcx: 'a> Visitor<'tcx> for CollectFunctionInstrumentationPoints<'a, 't
                 }
             }
 
-            if place.is_indirect() && !context.is_mutating_use() {
+            if place.is_indirect()
+                && !context.is_mutating_use()
+                // reborrows do not apply, i.e. _2 = &(*_1)
+                && !matches!(self.assignment, Some((_, Rvalue::Ref(..))))
+                // copying address of dereferenced value does not apply, i.e. _2 = &raw (*_1)
+                && !matches!(self.assignment, Some((_, Rvalue::AddressOf(..))) if base_ty.is_region_ptr())
+            {
```

Comment:
Maybe refactor this into a separate method.

Reply:
I'm going to save this for a PR for refactoring, which will follow before any new features.